### PR TITLE
Migrate doc from RTD to gh-pages

### PR DESCRIPTION
**Pull request by liruilong940607**
Workflow action for gh-pages.

New doc link: https://nerfstudio-project.github.io/nerfstudio

Need to update DNS (@tancik) and CNAME (in this PR @liruilong940607).

### CODE REVIEW

File: .readthedocs.yaml
Comment (tancik) [inline .readthedocs.yaml:1]: Delete this in followup PR

**Review by tancik — COMMENTED** [review]
Lets first set this up, then delete rtd once it is working.

File: .readthedocs.yaml
Comment (liruilong940607) [reply .readthedocs.yaml:1]: ok makes sense

**Comment by liruilong940607** [comment]
Fail on warning is tested:

https://github.com/nerfstudio-project/nerfstudio/actions/runs/6281838816/job/17060658890

**Comment by liruilong940607** [comment]
+ vis @brentyi @jake-austin

**Review by tancik — APPROVED** [review]
LGTM. Thanks for migrating!

**Pull request merged**